deepchecks/nlp/utils/text_properties.py

```diff
@@ -193,6 +193,38 @@ def formality(raw_text: Sequence[str], device: Optional[int] = None) -> List[flo
     classifier = get_transformer_pipeline('formality', model_name, device=device)
     return [x['score'] if x['label'] == 'formal' else 1 - x['score'] for x in classifier(raw_text)]
 
+def lexical_density(raw_text: Sequence[str]) -> List[str]:
+    """
+    Lexical density is the percentage of unique words in a given text. For more 
+    information: https://en.wikipedia.org/wiki/Lexical_density
+
+    Returns a list of floats of lexical density.
+    """
+    result = []
+    for text in raw_text:
+        try:
+            total_words = len(textblob.TextBlob(text).words)
+            total_unique_words = len(set(textblob.TextBlob(text).words))
+            lexical_density = round(total_unique_words * 100 / total_words, 2)
+            result.append(lexical_density)
+        except:
+            result.append(np.nan)
+    return result
+
+def unique_noun_count(raw_text: Sequence[str]) -> List[str]:
+    """Returns list of integers of number of unique noun words in the text"""
+    result = []
+    for text in raw_text:
+        try:
```

Comment:
try except is a time consuming operation in python, better to use a if condition:

if not pd.isna(text):

Reply:
Done